File: src/user-input/package-managers.ts
// Constants
import { PACKAGE_MANGERS } from '@/constants/package-mangers'

// Third-Party libraries
import inquirer from 'inquirer'

// Types
import type { PackageManager } from '@/types/package-mangers'

// NodeJS
import process from 'node:process'

// Utils
import { writeMessage } from '@/utils/console'
import { getErrorMessage } from '@/utils/errors'

export async function getPackageManger(): Promise<PackageManager> {
  try {
    const { packageManager } = await inquirer.prompt({
      type: 'rawlist',
      choices: PACKAGE_MANGERS,
      message: 'Which package manager do you wanna use?',
      name: 'packageManager'
    })

    return packageManager
  } catch (error) {
    writeMessage({
      type: 'error',
      message: getErrorMessage('PackageManagerSelection')
    })
    process.exit(1)
  }
}

File: src/user_input/package-managers.go
package userinput

import (
	"os"
	"stylelintbc/src/constants"
	"stylelintbc/src/types"
	"stylelintbc/src/utils"

	"github.com/AlecAivazis/survey/v2"
)

func GetPackageManager() types.PackageManager {
	var questions = []*survey.Question{
		{
			Name: "packageManger",
			Prompt: &survey.Select{
				Options: constants.PackageManagers,
				Message: "Which package manager do you wanna use?",
				Default: "npm",
			},
		},
	}

	answers := struct {
		PackageManager string `survey:"packageManger"`
	}{}

	err := survey.Ask(questions, &answers)

	if err != nil {
		utils.WriteMessage(utils.WriteMessageProps{
			Type:    "error",
			Message: string(utils.GetErrorMessage("PackageManagerSelection")),
		})

		os.Exit(1)
	}

	return types.PackageManager(answers.PackageManager)
}
